### DISCUSSION TITLE
Android's new photo picker requirement

### DISCUSSION THREAD

**Discussion by Ender22** [airsdk/Adobe-Runtime-Support#3489] [Q&A]
During my last app update, I received this warning from the Google Play store:

![android-photo-picker](https://github.com/user-attachments/assets/acd3c804-2d9e-4c30-a5bc-d865795fff3b)

My app is not a gallery app, and we just use photos occasionally allowing users to upload a profile photo or to upload a photo/video to the in-app forums so I believe I do not match the requirement of 'core functionality of editing, managing, and maintaining photos.'

Google says that apps using these permissions will soon need to go through a secondary review to prove the above. 

I received this warning because I am using the CameraRollExtended ane from the lovely people at distriqt which requires the 2 permissions above. I will ask Distriqt if they know any way around this issue but in the meantime, I'd like to know if the flash.media.cameraroll also requires these permissions? Does it use the Android photo picker behind the scenes? 

Does anyone have any other info on this?

Thank you,

**Comment by ajwfrost** [comment]
Using `flash.media.CameraRoll` to pick an image on Android will launch an intent with `ACTION_PICK` for an "image/*" type ...
https://developer.android.com/reference/android/content/Intent#ACTION_PICK
and I'm pretty sure we checked this when all the scoped storage stuff came in, so you should be able to use the result events to get at the image data..

There's a `requestPermission()` method on the class but I'm pretty sure this wouldn't need any specific permissions listed in the manifest. See e.g. https://stacktips.com/articles/writing-image-picker-using-intent-in-android

thanks

**Comment by marchbold** [comment]
This is likely an issue with some legacy code in the CameraRollExtended extension. I'll follow up with you in the [repository](https://github.com/distriqt/ANE-CameraRollExtended/issues/154) about what is required to resolve this one. Cheers
